components/chainhook-cli/src/service/mod.rs

```diff
@@ -116,22 +118,71 @@ impl Service {
         if !hord_disabled {
             // TODO: add flag
             // let _ = download_ordinals_dataset_if_required(&mut self.config, &self.ctx).await;
-        }
-
-        // Start chainhook event observer
-        let context_cloned = self.ctx.clone();
-        let event_observer_config_moved = event_observer_config.clone();
-        let observer_command_tx_moved = observer_command_tx.clone();
-        let _ = hiro_system_kit::thread_named("Chainhook event observer").spawn(move || {
-            let future = start_event_observer(
-                event_observer_config_moved,
-                observer_command_tx_moved,
-                observer_command_rx,
-                Some(observer_event_tx),
-                context_cloned,
+            info!(
+                self.ctx.expect_logger(),
+                "Ordinal indexing is enabled by default, checking index... (use --no-hord to disable ordinals)"
             );
-            let _ = hiro_system_kit::nestable_block_on(future);
-        });
+
+            if let Some((start_block, end_block)) = should_sync_hord_db(&self.config, &self.ctx)? {
+                if start_block == 0 {
+                    info!(
+                        self.ctx.expect_logger(),
+                        "Initializing hord indexing from block #{}", start_block
+                    );
+                } else {
+                    info!(
+                        self.ctx.expect_logger(),
+                        "Resuming hord indexing from block #{}", start_block
+                    );
+                }
+
+                let (tx, rx) = channel();
+
+                let mut moved_event_observer_config = event_observer_config.clone();
+                let moved_ctx = self.ctx.clone();
+
+                let _ = hiro_system_kit::thread_named("Initial predicate processing")
+                    .spawn(move || {
+                        if let Some(mut chainhook_config) =
+                            moved_event_observer_config.chainhook_config.take()
+                        {
+                            for bitcoin_predicate in chainhook_config.bitcoin_chainhooks.iter_mut()
+                            {
+                                bitcoin_predicate.enabled = false;
+                            }
+                            let mut bitcoin_predicates_ref: Vec<
+                                &chainhook_sdk::chainhooks::types::BitcoinChainhookSpecification,
+                            > = vec![];
+                            for bitcoin_predicate in chainhook_config.bitcoin_chainhooks.iter() {
+                                bitcoin_predicates_ref.push(bitcoin_predicate);
+                            }
```

Comment:
Would it be possible for us just to loop once here?

```suggestion
                            let mut bitcoin_predicates_ref: Vec<
                                &chainhook_sdk::chainhooks::types::BitcoinChainhookSpecification,
                            > = vec![];
                            for bitcoin_predicate in chainhook_config.bitcoin_chainhooks.iter_mut()
                            {
                                bitcoin_predicate.enabled = false;
                                bitcoin_predicates_ref.push(bitcoin_predicate);
                            }
```

Reply:
Addressed in https://github.com/hirosystems/chainhook/pull/307/commits/ba662e5149ecc9663e50c2f4cf0d1ba49e5486ee